### DISCUSSION TITLE
updater selection design questions

### DISCUSSION THREAD

**Discussion by prestwich** [celo-org/optics-monorepo#597] [Ideas]
Need designs for:
- [ ] updater rotation 
    - [x] when is an updater rotated?
    - [x] why?
    - [ ] how do we updater handle failure?
    - [ ] can we have round robin updates?
- [ ] replica notification
    - [ ] destination message that all replicas process?
- [ ] bond management
    - [ ] how is bond size set?
    - [ ] how is the bond custodied
- [ ] can we have synchrony assumptions?
    - [ ] should we embed home chain timestamp in messages?

**Comment by anna-carroll** [comment]
- can anyone be an Updater as long as they post a bonded stake?

**Comment by anna-carroll** [comment]
**Related**: unhappy case design questions #66

**Comment by anna-carroll** [comment]
@prestwich can you elaborate on "can we have synchrony assumptions" -- not sure what this means / the implications

**Reply by prestwich** [reply]
can we assume that messages get processed in a reasonable amount of time and that messages generally arrive in the order they were sent

**Comment by yourbuddyconner** [comment]
Related to updater rotation: https://github.com/celo-org/optics-monorepo/issues/610
